### PR TITLE
Jordy add email controller unit tests

### PR DESCRIPTION
**Pull request by j-corp-25**
# Description
Emailcontroller unit tests

## Related PRS (if any):
N/A
…

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. npm run test email -- --verbose
4. Check tests are passing

## Screenshots or videos of changes:

Screenshot 2024-09-21 at 7 14 50 PM

### CODE REVIEW

**Review by ArtemisNyx3 — DISMISSED** [review]
The emojis in the .md files are wrong in many cases 
eg: using a ❌ for positive cases or ✅ for negative cases

Can  you please fix them

**Review by chu-ziyu — APPROVED** [review]
I have run the unit test and all passed.
Screenshot 2024-09-24 at 4 28 10 PM

**Review by manikittu810 — APPROVED** [review]
Hello @j-corp-25 
Everything is working as mentioned.
Screenshot 2024-09-24 at 6 53 19 PM

Thanks.

**Comment by tharunaa-1** [comment]
All the test cases has been passed.
image

**Review by Lexie5212 — APPROVED** [review]
I have run the unit test, and all tests passed!
Screenshot 2024-09-25 at 13 18 35

**Comment by geetamatkar** [comment]
Hi @j-corp-25 ,
Tested this PR. All test cases have passed.

PR1114

**Review by nikhilpittala16 — APPROVED** [review]
I have tested the PR and all the test cases have passed.
Screenshot (207)

**Review by RishithaMamidala — APPROVED** [review]
@j-corp-25 
All unit tests for the emailController passed, including successful email sending, handling errors in email subscription updates, and proper validation for token and email in subscription functions. No issues were found.
PR 1114

**Review by ziyanw52 — APPROVED** [review]
I have run and test, all the tests passed.
341727472220_ pic_hd

**Review by ishangoel17 — COMMENTED** [review]
All test cases are passing. Below is the screenshot for reference:
Capture1114

**Review by Muzammil603 — APPROVED** [review]
https://docs.google.com/document/d/1Q-dOnKW8dRLsCbAVuwJHiXWrshETyQm7edZX2g6SrRA/edit
Results as expected

**Review by anirudhsk2107 — APPROVED** [review]
The tests have passed 

Screen Shot 2024-10-02 at 12 11 20 AM

**Comment by pallavithorat** [comment]
All test cases have successfully passed for this PR! Screenshot attached for reference.
PR #1114

**Comment by SatyashanthiT** [comment]
PR1114

Ran the provided command, and all EmailController unit tests passed successfully. Functionality is working as required.

File: src/controllers/emailController.spec.js [hunk 1]
```diff
@@ -0,0 +1,148 @@
+const { mockReq, mockRes, assertResMock } = require('../test');
+const emailController = require('./emailController');
+const jwt = require('jsonwebtoken');
+const EmailSubcriptionList = require('../models/emailSubcriptionList');
```
Comment (Abi-Liu) [inline src/controllers/emailController.spec.js:4]: Delete unused variable

File: src/controllers/emailController.spec.js [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+};
+describe('emailController Controller Unit tests', () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe('sendEmail function', () => {
+  test('should send email successfully', async () => {
+    const { sendEmail } = makeSut();
+    const mockReq = {
```
Comment (Abi-Liu) [inline src/controllers/emailController.spec.js:42]: This is the right idea, but you need to be careful not to reset the entire `mockReq` object. Instead you want to just append on the fields you need:
`mockReq.body.to = "asdf@asdf.com"`

This is ok for the unit tests, but if you do this for the integration tests things will start to break.

**Review by Abi-Liu — DISMISSED** [review]
Looks good! There's just a small issue with how you're setting the mockReq variable

**Review by jinxyou — APPROVED** [review]
Screenshot 2024-10-04 222012
All test cases have passed.

**Review by neeharikakoniki — APPROVED** [review]
The unit tests for the `EmailController` were thoroughly implemented, and all tests passed successfully.
image

**Review by lucy-xfl — APPROVED** [review]
image
Everything work well on my side

**Comment by mianmiantea2019** [comment]
Everything work well on my side

image

**Comment by j-corp-25** [comment]
> The emojis in the .md files are wrong in many cases eg: using a ❌ for positive cases or ✅ for negative cases
> 
> Can you please fix them

The checkmarks  ✅  were for cases I covered and  ❌ were for cases I didnt cover

**Review by samarth9201 — APPROVED** [review]
Confirmed that all the tests have passed

image

**Comment by one-community** [comment]
Thank you all, merging!

**Pull request merged**